modules/local/presto/presto_parseheaders_primers.nf

```diff
@@ -1,37 +1,38 @@
-include { initOptions; saveFiles; getSoftwareName } from '../functions'
-
-params.options = [:]
-def options    = initOptions(params.options)
-
 process PRESTO_PARSEHEADERS_PRIMERS {
     tag "$meta.id"
     label "process_low"
 
-    publishDir "${params.outdir}",
-        mode: params.publish_dir_mode,
-        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
-
-    conda (params.enable_conda ? "bioconda::presto=0.6.2=py_0" : null)              // Conda package
-    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
-        container "https://depot.galaxyproject.org/singularity/presto:0.6.2--py_0"  // Singularity image
-    } else {
-        container "quay.io/biocontainers/presto:0.6.2--py_0"                        // Docker image
-    }
+    conda (params.enable_conda ? "bioconda::presto=0.7.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/presto:0.7.0--pyhdfd78af_0' :
+        'quay.io/biocontainers/presto:0.7.0--pyhdfd78af_0' }"
 
     input:
     tuple val(meta), path(reads)
 
     output:
     tuple val(meta), path("*_reheader-pass.fastq"), emit: reads
+    path "versions.yml" , emit: versions
 
     script:
+    def args = task.ext.args ?: ''
     if (params.cprimer_position == "R1") {
         """
-        ParseHeaders.py copy -s $reads -o "${reads.baseName}_reheader-pass.fastq" -f PRCONS PRCONS --act first last -k C_PRIMER V_PRIMER
+        ParseHeaders.py copy -s $reads -o "${reads.baseName}_reheader-pass.fastq" -f $args --act first last -k C_PRIMER V_PRIMER
```

Comment:
Does the printing of the version need to be duplicated here?

Reply:
yes, because only one or the other script section is executed here depending on the if clause